### PR TITLE
Order the check_https_domains output

### PR DESCRIPTION
**Pull request by lamaral**
The check_https_domains now outputs the smallest amount of days to a
certificate expiration on first output.
On the additional lines, it shows each certificate expiration date,
ordered ascending by the amount of days remaining.

### CODE REVIEW

File: src/check_https_domains.py [hunk 1]
```diff
@@ -49,59 +51,68 @@ def get_domains():
...
+    print(message)
+    sys.exit(state)
 
 
 def main():
     domains = get_domains()
-    check_domains(domains, args.ip, args.hostname)
-    exit()
+    if not (domains and domains != ['$_HOSTDOMAINS$']):
+        return_result(3, 'UNKNOWN - No domain found for host: {}, ip: {}'
```
Comment (TheFRedFox) [inline src/check_https_domains.py]: Apply Multiline style code.

File: src/check_https_domains.py [hunk 2]
```diff
@@ -49,59 +51,68 @@ def get_domains():
...
+            output.append(
+                'CRITICAL - There are certificates expired for {} days'.format(
+                    -expirations[0]['remaining'].days))
+        else:
+            output.append(
+                'CRITICAL - There are certificates expiring in {} days'.format(
+                    expirations[0]['remaining'].days))
+    elif expirations[0]['remaining'] <= timedelta(days=warn):
+        state = 1
+        output.append(
```
Comment (TheFRedFox) [inline src/check_https_domains.py]: Apply Multiline style code.

File: src/check_https_domains.py [hunk 3]
```diff
@@ -49,59 +51,68 @@ def get_domains():
...
+
+    if expirations[0]['remaining'] <= timedelta(days=crit):
+        state = 2
+        if expirations[0]['remaining'] < timedelta(0):
+            output.append(
+                'CRITICAL - There are certificates expired for {} days'.format(
+                    -expirations[0]['remaining'].days))
+        else:
+            output.append(
+                'CRITICAL - There are certificates expiring in {} days'.format(
```
Comment (TheFRedFox) [inline src/check_https_domains.py]: Apply Multiline style code.

File: src/check_https_domains.py [hunk 4]
```diff
@@ -49,59 +51,68 @@ def get_domains():
...
+        expirations.append({'remaining': remaining, 'common_name': common_name,
+                            'domain': domain_tmp, 'not_after': not_after})
+
+    expirations.sort(key=lambda x: x['remaining'])
+
+    if expirations[0]['remaining'] <= timedelta(days=crit):
+        state = 2
+        if expirations[0]['remaining'] < timedelta(0):
+            output.append(
+                'CRITICAL - There are certificates expired for {} days'.format(
```
Comment (TheFRedFox) [inline src/check_https_domains.py]: Apply Multiline style code.

File: src/check_https_domains.py [hunk 5]
```diff
@@ -21,25 +21,27 @@
...
 import sys
-import subprocess
-import shlex
 from argparse import ArgumentParser, RawTextHelpFormatter
+from datetime import datetime, timedelta
+from dateutil.parser import parse
+from dateutil.tz import tzutc
+from OpenSSL import crypto
 
 parser = ArgumentParser(description='Check domains',
```
Comment (TheFRedFox) [inline src/check_https_domains.py]: Actually the parser should be wrapped inside an own function and be called in `main` function and then just the required arguments should be passed to other functions like `get_domains` or `get_check_result`.

File: src/check_https_domains.py [hunk 6]
```diff
@@ -49,59 +51,68 @@ def get_domains():
...
-        if (returncode > state and state != 2) or returncode == 2:
-            state = returncode
+def get_check_result(domains, ip):
+    output = []
+    expirations = []
+    for domain_tmp in domains:
+        domain = domain_tmp.replace('*', 'www', 1)
+        conn = ssl.create_connection((ip, 443))
+        context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+        sock = context.wrap_socket(conn, server_hostname=domain)
```
[the remark this answers is not in the document]
Comment (lamaral) [reply src/check_https_domains.py]: It does. Good idea!

File: src/check_https_domains.py [hunk 7]
```diff
@@ -29,74 +29,106 @@
...
         if expirations[0]['remaining'] < timedelta(0):
             output.append(
                 'CRITICAL - There are certificates expired for {} days'.format(
-                    -expirations[0]['remaining'].days))
+                    -expirations[0]['remaining'].days
+                )
+            )
         else:
             output.append(
                 'CRITICAL - There are certificates expiring in {} days'.format(
```
Comment (TheFRedFox) [inline src/check_https_domains.py]:
``` python
output.append(
    'CRITICAL - There are certificates expiring in {} days'
    .format(expirations[0]['remaining'].days)
)
```

**Pull request merged**